server.go

```diff
@@ -322,13 +331,63 @@ func (s *ServerImpl) ListDeploys() ([]*Deploy, error) {
 			Id:      fmt.Sprintf("%s-%d", proc.Name, proc.Port),
 			Pid:     proc.Pid,
 			Port:    proc.Port,
+			Set:     proc.Port == setPort,
 			Tracked: false,
 		})
 	}
 	s.checkAllHealth(knownRunningDeploys)
 	return append(knownDeploys, unaccounted...), nil
 }
 
+func (s *ServerImpl) getPortMarkedAsSet() (int, error) {
+
+	url := fmt.Sprintf("http://localhost:%d/;csv", s.endPort)
+	resp, err := http.Get(url)
+	if err != nil {
+	  // Couldn't connect to haproxy don't modify the deploy list
+	  return -1, err
+	}
+	
+	defer resp.Body.Close()
+
+	records, err := csv.NewReader(resp.Body).ReadAll()
+	if err != nil {
+	  return -1, err
+	}
+
+	for i, row := range records {
+	  
+	  // skip header
+	  if i == 0 {
+	    continue
+	  } 
+
+	  // Find the first row that contains a Process name that 
+	  // looks like 'thing-PORT". 
+	  if strings.Contains(row[haProxyPxnameIndex], "thing-") {
```

Comment:
Moved this to a function in the haproxy module and created the helper functions as suggested